### PR TITLE
replace soundfile with librosa

### LINKED ISSUE #724: Speech recognition data loading and augmentation
## 🚀 Feature

https://github.com/PyTorchLightning/lightning-flash/blob/4ebc45dd74df50e0a8b8d8e92efb97c6c0b9f6cb/flash/audio/speech_recognition/data.py#L72

At the moment the audio is getting loaded by soundfile, if it would be replaced with librosa more audio types are supported and resampling is done by default.

### Motivation

More audio formats like MP3 would be supported.
Furthermore an augmentation of data would be cool for training, especially with low ressource domains or languages.

### Pitch

Replacing soundfile with librosa and integrating audiomentations library.

### Additional context

I am not familiar with lightning flash API, especially I dont know how to add augmentation for training only, when its the same method used for prediction and training.
I would like to add these two features, if you give some advice.
In the same step I could check adding Hubert, or replacing the hardcoded classes with automodel class, if it exists.

One more question:
What happens when I load an only pretrained model without feature extractor ? Does LF built its automatically ?

### PR DESCRIPTION
**Pull request by flozi00**
## What does this PR do?

This PR replaces soundfile with librosa to support more audio formats like .mp3 by default

Fixes #724

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by flozi00** [comment]
@ethanwharris

I have an strange error, while training on mp3 data everything works fine, but when using mp3 files to predict librosa is never called to load the audio. When using wav-files the print statement executes.


```
Traceback (most recent call last):
  File "f:\codes\python apps\lightning-flash\flash\core\data\batch.py", line 231, in forward
    samples = self.collate_fn(samples, metadata)
  File "C:\Users\flozi\anaconda3\envs\wav2vec\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "f:\codes\python apps\lightning-flash\flash\core\data\utils.py", line 178, in forward
    return self.func(*args, **kwargs)
  File "f:\codes\python apps\lightning-flash\flash\core\data\process.py", line 383, in collate
    return collate_fn(samples, metadata)
  File "f:\codes\python apps\lightning-flash\flash\audio\speech_recognition\collate.py", line 65, in __call__
    sampling_rates = [sample["sampling_rate"] for sample in metadata]
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred: 

Traceback (most recent call last):
  File "test-flash.py", line 15, in <module>
    print(model.predict(sample_path))
  File "f:\codes\python apps\lightning-flash\flash\core\model.py", line 247, in wrapper
    result = func(self, *args, **kwargs)
  File "f:\codes\python apps\lightning-flash\flash\core\model.py", line 418, in predict
    x = data_pipeline.worker_preprocessor(running_stage, collate_fn=dataloader.collate_fn)(x)
  File "C:\Users\flozi\anaconda3\envs\wav2vec\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "f:\codes\python apps\lightning-flash\flash\core\data\batch.py", line 233, in forward
    samples = self.collate_fn(samples)
  File "C:\Users\flozi\anaconda3\envs\wav2vec\lib\site-packages\torch\nn\modules\module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "f:\codes\python apps\lightning-flash\flash\core\data\utils.py", line 178, in forward
    return self.func(*args, **kwargs)
  File "f:\codes\python apps\lightning-flash\flash\core\data\process.py", line 383, in collate
    return collate_fn(samples, metadata)
  File "f:\codes\python apps\lightning-flash\flash\audio\speech_recognition\collate.py", line 65, in __call__
    sampling_rates = [sample["sampling_rate"] for sample in metadata]
TypeError: 'NoneType' object is not iterable
```

cause the audio never gets loaded an NoneType error returns

**Comment by ethanwharris** [comment]
Hey @flozi00 great work here! I fixed the MP3 predicting issue, just needed to add `.mp3` to the list of valid extensions in the `SpeechRecognitionPathsDataSource`. Should all be working now :smiley:

**Review by ethanwharris — APPROVED** [review]
Awesome, LGTM :smiley:

File: flash/audio/speech_recognition/data.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def deserialize(self, sample: Any) -> Dict:
         encoded_with_padding = (sample + "===").encode("ascii")
         audio = base64.b64decode(encoded_with_padding)
         buffer = io.BytesIO(audio)
-        data, sampling_rate = sf.read(buffer)
+        data, sampling_rate = librosa.load(buffer, sr=16000)
```
Comment (SeanNaren) [inline flash/audio/speech_recognition/data.py:51]: can we expose the sample rate here at all?

Comment (ethanwharris) [reply flash/audio/speech_recognition/data.py]: Done :smiley:

**Pull request merged**